src/Functions/URL/decodeURLComponent.cpp

```diff
@@ -11,6 +11,31 @@ namespace ErrorCodes
     extern const int ILLEGAL_COLUMN;
 }
 
+static size_t encodeURL(const char * src, size_t src_size, char * dst, bool space_as_plus)
+{
+    char * dst_pos = dst;
+    for (size_t i = 0; i < src_size - 1; i++)
+    {
+        if ((src[i] >= '0' && src[i] <= '9') || (src[i] >= 'a' && src[i] <= 'z') || (src[i] >= 'A' && src[i] <= 'Z')
+            || src[i] == '-' || src[i] == '_' || src[i] == '.' || src[i] == '!' || src[i] == '~' || (src[i] >= '\'' && src[i] <= '*'))
```

Comment:
Please use formatting for this checks, as I understand this is part of specification from here https://en.wikipedia.org/wiki/Percent-encoding ?
Also does this `src[i] >= '\'' && src[i] <= '*'` handles `!	#	$	&	'	(	)	*	+	,	/	:	;	=	?	@	[	]` ?

Reply:
src[i] >= '\'' && src[i] <= '*' => * ' ( ) , i use formatting for this checks by Javascript's standard: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent ,it is: A-Z a-z 0-9 - _ . ! ~ * ' ( ), It has been corrected as:- _ . ~ from(RFC 3986:https://en.wikipedia.org/wiki/Percent-encoding)